tfhe/src/core_crypto/algorithms/seeded_lwe_ciphertext_decompression.rs

```diff
@@ -26,16 +27,20 @@ pub fn decompress_seeded_lwe_ciphertext_with_existing_generator<Scalar, OutputCo
     );
 
     let ciphertext_modulus = output_lwe.ciphertext_modulus();
-    assert!(ciphertext_modulus.is_compatible_with_native_modulus());
     let (mut output_mask, output_body) = output_lwe.get_mut_mask_and_body();
 
     // generate a uniformly random mask
     generator.fill_slice_with_random_mask_custom_mod(output_mask.as_mut(), ciphertext_modulus);
-    if !ciphertext_modulus.is_native_modulus() {
-        slice_wrapping_scalar_mul_assign(
-            output_mask.as_mut(),
-            ciphertext_modulus.get_power_of_two_scaling_to_native_torus(),
-        );
+    match ciphertext_modulus.kind() {
+        // Manage the specific encoding for non native power of 2
+        CiphertextModulusKind::NonNativePowerOfTwo => {
+            slice_wrapping_scalar_mul_assign(
+                output_mask.as_mut(),
+                ciphertext_modulus.get_power_of_two_scaling_to_native_torus(),
+            );
+        }
+        // Nothing to do
+        CiphertextModulusKind::Native | CiphertextModulusKind::Other => (),
     }
```

Comment:
I think it would be better to have `output_mask` written to only once
Could be done cleanly (IMO) by adding a `scale` argument to `fill_slice_with_random_mask_custom_mod`

Reply:
Applies to `decompress_seeded_lwe_ciphertext_list_with_existing_generator`  and `par_decompress_seeded_lwe_ciphertext_list_with_existing_generator` too

Reply:
I don't think it's possible with the current CSPRNG design and randomness design, plus this way you have a tight loop around the CSPRNG and a scalar mul which can easily be vectorized by the compiler, so don't think I'll change it (the CSPRNG fills a full slice and it's a design I don't want to update for now this is some of the oldest legacy code and I don't thing it brings value to refactor at this point in time)

Reply:
Ok